test/backlinks_test.dart

```diff
@@ -0,0 +1,139 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'package:realm_common/realm_common.dart';
+import 'package:test/expect.dart';
+
+import '../lib/realm.dart';
+import 'test.dart';
+
+part 'backlinks_test.g.dart';
+
+@RealmModel()
+class _Source {
+  String name = 'source';
+  _Target? oneTarget;
+  List<_Target> manyTargets = [];
+}
+
+@RealmModel()
+class _Target {
+  @Backlink(#oneTarget)
+  late Iterable<_Source> oneToMany; // computed property, so must go last in generated class!
```

Comment:
Why not ask people to declare backlinks as `RealmResults<_Source>`? We're using results under the hood, so don't see much value in those being iterables in the model.

Reply:
You could say the same for `List`. 

However, they are generated as `RealmResults` on `Target`, so you have the full interface available, ie `target.oneToMany[0]`, `target.manyToMany.changes.listen(...`,  etc.

I think it match `List` and `Set` better. 

I did consider [UnmodifiableListView](https://api.dart.dev/stable/2.18.3/dart-collection/UnmodifiableListView/UnmodifiableListView.html), and can be persuaded to switch, but it is not perfect either.

Reply:
Aren't we using RealmList for the models? Or do you mean that we allow people to declare them either as List or RealmList?

Reply:
No. In the realm models we are using `List<T>`

Reply:
But in the generated classes we also have realm specific stuff like `freeze()`,  `changes`, etc. So the generated class overrides with `RealmList`. The same goes for back-links, they become `RealmResults<T>`.

Reply:
@blagoev The comment is their, because it is important that the generator re-orders these fields, so that the generated props go last. That is a requirement before handing the props over to core.

Reply:
Yeah, this is slightly different from the C# SDKs behavior, but I think that's fine. I got confused because I thought we're declaring properties as `RealmList`, but in reality we're doing it with `List`. Fine with resolving this on my end.

Reply:
Sorry for hijacking the thread here. 

I suggest we remove these comments from the tests. These serve no good in the tests here. I don't see how they help if I read the code. Should the next one that refactors do something about it.

Reply:
What if I change to:
```dart
@RealmModel()
class _Target {
  @Backlink(#oneTarget)
  late Iterable<_Source> oneToMany; // will be reordered by the generator to go after name

  String name = 'target';

  @Backlink(#manyTargets)
  late Iterable<_Source> manyToMany;
}
```
?

Reply:
Actually I would suggest we change the generator comment to be more clear. Something like `Realm Core requires computed properties at the end so we sort them at generation time versus doing it at runtime every time`.

Reply:
I have done the comment change in the generator. I have left a comment here in case someone looks at the generated class, and wonder why the fields are re-ordered.

Reply:
yeah but why would they care. I don't remember one time I needed to check the generated class for the order. and there is the comment in the generator

Reply:
I don't think users should care. It is more for the future me..